### ISSUE TITLE
Unable to launch AppHost after updating to Preview 5

### ISSUE THREAD

**Issue by g-martin772** [dotnet/aspire#3665]
I installed the newest SDK + Workloads and replaced the Aspire.Hosting package with the new Aspire.Hosting.AppHost package and fixed all compile errors, but when launching even a simplified setup like this I get the following failure "Unable to create Session". 

Code:
```cs
var redis = builder.AddRedis("redis");

var sql = builder.AddSqlServer("sql")
    .WithDataVolume()
    .AddDatabase("xDb");

var api = builder.AddProject<ProjApi>("api")
    .WithReference(sql)
    .WithReference(redis);

var blazor = builder.AddProject<ProjBlazor>("blazor")
    .WithReference(api)
    .WithReference(redis)
    .WithReference(sql);

builder.Build().Run();
```
Log Output:
```
info: Aspire.Hosting.DistributedApplication[0]
      Aspire version: 9.0.0-preview.3.24210.17+c2d83148f169bdf4980393a66eff2a28e94eb6c9
info: Aspire.Hosting.DistributedApplication[0]
      Distributed application starting.
info: Aspire.Hosting.DistributedApplication[0]
      Application host directory is: [...]ProjName.AppHost
info: Aspire.Hosting.DistributedApplication[0]
      Now listening on: http://localhost:15073
info: Aspire.Hosting.DistributedApplication[0]
      Distributed application started. Press Ctrl+C to shut down.
fail: Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler[0]
      run session could not be started  {"Executable": {"name":"blazor-9hgdkgg"}, "Reconciliation": 7, "Status": "400 Bad Request", "Body": "\"Unable to create a session\""}
fail: Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler[0]
      run session could not be started  {"Executable": {"name":"api-9to76dg"}, "Reconciliation": 6, "Status": "400 Bad Request", "Body": "\"Unable to create a session\""}
```

Dashboard: 
![image](https://github.com/dotnet/aspire/assets/95913933/ded0d6b9-6457-4c15-a820-6728484d8df6)

dotnet workload list
```
Installed Workload Id      Manifest Version                                Installation Source                     
-------------------------------------------------------------------------------------------------------------------
aspire                     9.0.0-preview.3.24210.17/9.0.100-preview.1      SDK 9.0.100-preview.3, VS 17.10.34607.79
maui-windows               9.0.0-preview.3.10457/9.0.100-preview.3         VS 17.10.34607.79                       
maccatalyst                17.2.9433-net9-p3/9.0.100-preview.3             VS 17.10.34607.79                       
ios                        17.2.9433-net9-p3/9.0.100-preview.3             VS 17.10.34607.79                       
android                    34.99.0-preview.3.231/9.0.100-preview.3         VS 17.10.34607.79                


Use `dotnet workload search` to find additional workloads to install.
```

**Comment by g-martin772** [comment]
Update: This only happens when running from inside of Rider using the IDE integration, using the CLI it actually starts fine so this is probably a Rider issue but I still have no idea what is causing it. Any ideas?

**Comment by asimmon** [comment]
I don't see `8.0.0-preview.5.24201.12` showing up in your logs and screenshots. Is it normal that you are targeting .NET 9 (instead of .NET 8)?

**Comment by timheuer** [comment]
> Rider issue but I still have no idea what is causing it since preview 4 worked fine

In Preview 5 there were likely enough changes in the appmodel and IDE protocol adjustments (and security) that Rider probably needs to adapt to and hasn't just yet since we're still in Preview (and their support also in Preview).  Thanks for confirming all works in CLI as then indeed it's likely an IDE protocol point-in-time with Rider.

**Comment by g-martin772** [comment]
It is fixed with the new preview 6 version and the 6.0 Azure plugin for rider, so I will just close this.
